### ISSUE TITLE
Enhancement: remove subscriptions by owner

### ISSUE THREAD

**Issue by zsau** [slazyk/Observable-Swift#20]
What do you think about adding a method like `Event.remove(owner owner:AnyObject)`? It would eliminate the need to store subscriptions in an instance var in the common case. I can't implement it in an extension, because the list of subscriptions is marked `internal`.

**Comment by travisgriggs** [comment]
I would like to second this request as well.

**Comment by marchy** [comment]
Agreed, this would indeed greatly simplify things, as well as take steps towards reducing the chance of memory leaks from loose registrations as well. It's also nice not to require callbacks to be specified in a method. Sometimes in-line closures/lambdas is all you want/need.

**Comment by zsau** [comment]
Is this something you're interested in, @slazyk? Will gladly provide a pull request if so.
